cmake/menuconfig.cmake

```diff
@@ -74,6 +74,24 @@ add_custom_target(
           ${NUTTX_DEFCONFIG}
   WORKING_DIRECTORY ${NUTTX_DIR})
 
+add_custom_target(
+  oldconfig
+  COMMAND ${CMAKE_COMMAND} -E env ${KCONFIG_ENV} oldconfig
+  COMMAND ${CMAKE_COMMAND} -E remove -f
```

Comment:
why need

Reply:
Both the `oldconfig` and `olddefconfig` commands change the `.config` file, but that doesn't automatically regenerate `include/nuttx/config.h`.

Looking at [this comment](https://github.com/apache/nuttx/blob/fe642f1cfddfd89a713690bc67be614d402f4dcd/cmake/menuconfig.cmake#L21-L22), it seems like proper dependency tracking to regenerate `include/nuttx/config.h` when `.config` changes is not yet implemented. I took both the commands you commented about from `menuconfig`.

This command specifically removes `include/nuttx/config.h` to force it being regenerated at the next build.

Reply:
@xuxin930 could you review this change?